plugins/jobs/src/js/JobDelete.js

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/map";
+import "rxjs/add/operator/switchMap";
+import "rxjs/add/operator/do";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/startWith";
+import "rxjs/add/operator/catch";
+
+import gql from "graphql-tag";
+
+import JobDeleteModal from "./components/JobDeleteModal";
+import defaultSchema from "./data/JobModel";
+
+const deleteJob = gql`
+  mutation {
+    deleteJob(id: $jobId, stopCurrentJobRuns: $stopCurrentJobRuns) {
+      jobId
+    }
+  }
+`;
+
+function executeDelete({ jobId, stopCurrentJobRuns, onSuccess, errorMsg }) {
+  return graphqlObservable(deleteJob, defaultSchema, {
+    jobId,
+    stopCurrentJobRuns
+  })
+    .map(_ => ({ done: true, stopCurrentJobRuns, errorMsg }))
+    .do(_ => onSuccess())
+    .startWith({ done: false, stopCurrentJobRuns, errorMsg });
+}
+
+function deleteOperation() {
+  const deleteSubject$ = new Subject();
+  const delete$ = deleteSubject$.switchMap(executeDelete).catch(error => {
+    return delete$.startWith({
+      errorMsg: error.response.message,
+      done: true
+    });
+  });
+
+  return {
+    delete$,
+    deleteHandler: (jobId, stopCurrentJobRuns, onSuccess, errorMsg) => {
+      deleteSubject$.next({ jobId, stopCurrentJobRuns, onSuccess, errorMsg });
+    }
+  };
+}
+
+function errorIsTaskCurrentRunning(errorMsg) {
+  return /stopCurrentJobRuns=true/.test(errorMsg);
+}
+
+const JobDelete = componentFromStream(prop$ => {
+  const { delete$, deleteHandler } = deleteOperation();
```

Comment:
We would create a new subject every time the props change, does this really make sense?
(I see this in multiple occasions but choose to only comment it here to discuss it more easily)

Reply:
I can verify that. But I think that is not the case. componentFromStream will be called only once to generate the component. I believe this function will be called on construction or mounting. 

Notice that this is outside of the `prop$.combine.map..` which means that it does not get called on every property refresh.

Reply:
Oh okay, please verify. If it is like you say then it's totally fine

Reply:
It is only called once.